### PR TITLE
[DEV-5551][DEV-5549] Fix wording and some award counts

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**High level description:**

Update unlinked text to match AC and use Award Count when determining missing awards.

**Technical details:**

Refactored some to minimize calls to `/api/v2/disaster/award/amount/` since the endpoint is causing a lot of trouble. Instead of multiple calls for the same endpoint there is one call to handle the "All Awards" tab across different visualizations and that one call is stored in the Redux store.

**JIRA Ticket:**
[DEV-5551](https://federal-spending-transparency.atlassian.net/browse/DEV-5551)
[DEV-5549](https://federal-spending-transparency.atlassian.net/browse/DEV-5549)

**Mockup:**
N/A

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, helper functions, and models `if applicable`
- [x] All `componentWillReceiveProps` and `componentWillMount` in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3277](https://federal-spending-transparency.atlassian.net/browse/DEV-3277)

Reviewer(s):
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx [hunk 1]
```diff
@@ -61,6 +61,13 @@ const SpendingByCFDA = () => {
         Analytics.event({ category: 'COVID-19 - Award Spending by CFDA', action: `${activeTab} - click` });
     };
 
+    // Keep track of previous tab to prevent duplicate requests in summary
+    const prevTabRef = useRef();
+    useEffect(() => {
+        prevTabRef.current = activeTab;
```
Comment (MaxwellKendall) [inline src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:67]: This duplicates our state. A `ref` [should be used sparingly for `mutuable` data](https://reactjs.org/docs/hooks-reference.html#useref). Typically this means a DOM Node. This is how we use it. In this case, we're putting immutable data that duplicates a property already in state inside a ref. We should just use the state property here.

File: src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx [hunk 2]
```diff
@@ -61,6 +61,13 @@ const SpendingByCFDA = () => {
...
+    });
+    const prevTab = prevTabRef.current;
```
Comment (MaxwellKendall) [inline src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:69]: In this case, we're setting the `prevTab` to equal the value of a ref, which itself is the value of a property in state called `activeTab`. So `prevTab` equals `activeTab`? This is self-refuting and confusing. Not sure what exactly we're trying to do here, but I think we're `trying` to capture stale state? This shouldn't be necessary.

File: src/js/containers/covid19/SummaryInsightsContainer.jsx [hunk 3]
```diff
@@ -43,53 +45,71 @@ const SummaryInsightsContainer = ({
     const [numberOfAwards, setNumberOfAwards] = useState(null);
     const initialInFlightState = overviewData.map((d) => d.type);
     const [inFlightList, , removeFromInFlight, resetInFlight] = useInFlightList(initialInFlightState);
-    const defCodes = useSelector((state) => state.covid19.defCodes);
+    const { defCodes, allAwardTypeTotals } = useSelector((state) => state.covid19, isEqual);
 
     useEffect(() => {
+        if (assistanceOnly && (activeTab === 'all' && prevTab === 'all')) {
```
Comment (MaxwellKendall) [inline src/js/containers/covid19/SummaryInsightsContainer.jsx:51]: This reminds me of using component lifecycle methods. W/ useEffect, we have to use a different mental model. I'm not confident this prevTab value is going to be reliable.

Also, are we making duplicative requests to this endpoint in production?  If so, that could be a bug you can scope outside this delivery.

File: src/js/helpers/covid19Helper.js [hunk 4]
```diff
@@ -164,10 +164,13 @@ export const getTotalSpendingAbbreviated = (totalSpending) => {
...
-export const areCountsDefined = (counts) => Object.keys(counts).reduce((acc, tab) => {
-    if (acc === null) return acc;
-    return counts[tab];
-}, true);
+export const areCountsDefined = (counts) => {
+    const countTabs = Object.keys(counts);
+    if (countTabs.length === 0) {
+        return false;
+    }
+    return countTabs.reduce((acc, tab) => acc && counts[tab] !== null, true);
```
Comment (MaxwellKendall) [inline src/js/helpers/covid19Helper.js:172]: Can we add a test case to cover the condition where this was returning `falsy`? This would just mean adding an `it` block to the existing tests where one of the tab counts is zero; preferably the last property in the tabCounts object like

```javascript
const mockTabCounts = { test1: 5, test2: 1, test3: 0 }
```

File: src/js/containers/covid19/Covid19Container.jsx [hunk 5]
```diff
@@ -121,14 +127,39 @@ const Covid19Container = () => {
...
         };
+        const getAllAwardTypesAmount = async () => {
+            const params = {
+                filter: {
+                    def_codes: defCodes.map((code) => code.code)
+                }
+            };
+            awardAmountRequest.current = fetchAwardAmounts(params);
+            awardAmountRequest.current.promise
+                .then((res) => {
```
Comment (MaxwellKendall) [inline src/js/containers/covid19/Covid19Container.jsx:138]: to pass the eslint camelcase r ule:

```suggestion
                .then(({
                  data: {
                    obligation,
                    outlay,
                    award_count: awardCount,
                    face_value_of_loan: faceValueOfLoan
                  }
                }) => {
```

File: src/js/containers/covid19/SummaryInsightsContainer.jsx [hunk 6]
```diff
@@ -43,53 +45,71 @@ const SummaryInsightsContainer = ({
...
+                    params.filter.award_type = 'assistance';
+                }
+                awardAmountRequest.current = fetchAwardAmounts(params);
+                awardAmountRequest.current.promise
+                    .then((res) => {
+                        setAwardObligations(res.data.obligation);
+                        setAwardOutlays(res.data.outlay);
+                        setNumberOfAwards(res.data.award_count);
+
+                        /* eslint-disable camelcase */
```
Comment (MaxwellKendall) [inline src/js/containers/covid19/SummaryInsightsContainer.jsx:93]: ditto per above we can use object de-structuring to pass this test and we probably should to make it more readable as to what these properties on the api actually are.

**Review by MaxwellKendall — CHANGES_REQUESTED** [review]

File: src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx (hunk 1 above)
Comment (sethstoudenmier) [reply src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:67]: This particular use case I think makes sense to not use state sense I am trying to track the previous value of the state: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

To use state would mean adding an additional update in the code for each state update when you can let this ref handle tracking that.

File: src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx (hunk 2 above)
Comment (sethstoudenmier) [reply src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:69]: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

File: src/js/containers/covid19/SummaryInsightsContainer.jsx (hunk 3 above)
Comment (sethstoudenmier) [reply src/js/containers/covid19/SummaryInsightsContainer.jsx:51]: As mentioned in our quick call, this particular duplicate call was noticed as a result of "all" for CFDA being considered something different than the other tables. Want to make sure we aren't sending a second request for CFDA when an update to `allAwardTypeTotals` triggers this hook.

File: src/js/helpers/covid19Helper.js (hunk 4 above)
Comment (sethstoudenmier) [reply src/js/helpers/covid19Helper.js:172]: Will do

File: src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:67]: Oh my! I had no idea this was a convention recommended in the documentation. Nicely done sir.

Comment (sethstoudenmier) [reply src/js/components/covid19/assistanceListing/SpendingByCFDA.jsx:67]: I will say if this becomes a more common thing then we should create a custom hook, but right now this is just a one off use case.

**Review by MaxwellKendall — APPROVED** [review]
